libs/libc/pthread/pthread_setspecific.c

```diff
@@ -72,9 +74,23 @@
 
 int pthread_setspecific(pthread_key_t key, FAR const void *value)
 {
-  int ret = tls_set_value((int)key, (uintptr_t)value);
-  return ret < 0 ? -ret : 0;
+  FAR struct tls_info_s *info;
+
+  DEBUGASSERT(key < CONFIG_TLS_NELEM);
+  if (key < CONFIG_TLS_NELEM)
```

Comment:
What is the point of `DEBUGASSERT` + `if` the same condition?
Also do we need to check `key >= 0`?

Reply:
Compiler may complain the check(key >= 0) is redundant since key is an unsigned int.

Reply:
But I see
```
#ifndef __PTHREAD_KEY_T_DEFINED
typedef int pthread_key_t;
#define __PTHREAD_KEY_T_DEFINED 1
#endif
```
Are you using custom `pthread_key_t` definition as unsigned?

Reply:
Ok, my editor direct me to the wrong definition(glibc version). All code check `key >= 0` now.